### DISCUSSION TITLE
How do I use Keycloak with Jakarta 10 Glassfish?

### DISCUSSION THREAD

**Discussion by magical-heyrovsky** [keycloak/keycloak#23643] [Q&A]
I am trying something like this:

```
@OpenIdAuthenticationMechanismDefinition(
        providerURI = "${oidcConfig.issuerUri}",
        clientId = "${oidcConfig.clientId}",
        clientSecret = "${oidcConfig.clientSecret}",
        redirectURI = "${baseURL}/callback",
        redirectToOriginalResource = true,
        jwksReadTimeout = 5000
)

```
with `oidcConfig.issuerUri` = `https://my-keycloak.com/auth/realms/my-realm/.well-known/openid-configuration` and so on but I am getting 

> Jakarta Authentication: Exception during validateRequest                                                                                                                
> jakarta.ws.rs.ProcessingException: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to f
> ind valid certification path to requested target    $

Does anyone have a working example of Keycloak + Jakarta 10 Glassfish?

**Comment by justin-tay** [comment]
If you are using a self-signed certificate for Keycloak you would need to modify the trust store that Glassfish is using accordingly.
